jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/HttpOutput.java

```diff
@@ -562,7 +565,16 @@ public void close() throws IOException
                             _apiState = ApiState.BLOCKED;
                             _state = State.CLOSING;
                             blocker = _writeBlocker.callback();
-                            content = _aggregate != null && _aggregate.hasRemaining() ? _aggregate.getByteBuffer() : BufferUtil.EMPTY_BUFFER;
+                            aggregate = _aggregate;
+                            if (aggregate != null && aggregate.hasRemaining())
+                            {
+                                aggregate.retain();
+                                content = aggregate.getByteBuffer();
```

Comment:
Generally getting the ByteBuffer from the aggregate is an anti-pattern as we want to avoid unnecessary copies if the aggregate is internally retaining.

Can't the `content` be a RBB and that way you can use RBB.writeTo below?

Reply:
That can be done, and looking more closely at that code I think I figured what the problem is in https://github.com/jetty/jetty.project/issues/12721.

Let me change that.